src/ocean/meta/traits/Basic.d

```diff
@@ -400,6 +400,43 @@ unittest
 
 *******************************************************************************/
 
+template isReferenceType ( T )
+{
+    const isReferenceType =
+           isPointerType!(T)
+        || isFunctionType!(T)
```

Comment:
Just thought about it: `isPointerType` will detect function pointers so `is(T == delegate)` should be sufficient here.